src/main/java/com/google/impactdashboard/server/DataUpdater.java

```diff
@@ -1,39 +1,66 @@
 package com.google.impactdashboard.server;
 
+import com.google.cloud.logging.v2.LoggingClient.ListLogEntriesPagedResponse;
+import com.google.cloud.logging.v2.LoggingClient;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.impactdashboard.data.IAMBindingDatabaseEntry;
+import com.google.impactdashboard.data.project.ProjectIdentification;
 import com.google.impactdashboard.data.recommendation.Recommendation;
+import com.google.impactdashboard.database_manager.data_read.DataReadManager;
+import com.google.impactdashboard.database_manager.data_read.DataReadManagerFactory;
 import com.google.impactdashboard.database_manager.data_update.DataUpdateManager;
 import com.google.impactdashboard.database_manager.data_update.DataUpdateManagerFactory;
+import com.google.impactdashboard.server.api_utilities.IamBindingRetriever;
 import com.google.impactdashboard.server.api_utilities.LogRetriever;
+import com.google.impactdashboard.server.api_utilities.ProjectListRetriever;
 import com.google.impactdashboard.server.api_utilities.RecommendationRetriever;
+import com.google.logging.v2.LogEntry;
 
 import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
 import java.util.List;
 
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
 /** Class for updating the information in the database from the API. */
 public class DataUpdater {
 
   private final LogRetriever logRetriever;
   private final RecommendationRetriever recommendationRetriever;
+  private final IamBindingRetriever iamRetriever;
   private final DataUpdateManager updateManager;
+  private final DataReadManager readManager;
+  private final ProjectListRetriever projectRetriever;
+  private final boolean manualUpdate;
 
   @VisibleForTesting
   protected DataUpdater(LogRetriever logRetriever, RecommendationRetriever recommendationRetriever,
-                      DataUpdateManager updateManager) {
+                      DataUpdateManager updateManager, DataReadManager readManager,
+                      IamBindingRetriever iamRetriever, ProjectListRetriever projectRetriever,
+                      boolean manualUpdate) {
     this.logRetriever = logRetriever;
     this.recommendationRetriever = recommendationRetriever;
     this.updateManager = updateManager;
+    this.readManager = readManager;
+    this.iamRetriever = iamRetriever;
+    this.projectRetriever = projectRetriever;
+    this.manualUpdate = manualUpdate;
   }
 
   /**
    * Static factory for creating a new instance of DataUpdater with new instances of LogRetriever
    * and RecommendationRetriever.
    * @return New instance of DataUpdater
    */
-  public static DataUpdater create() throws IOException {
+  public static DataUpdater create(boolean manualUpdate) 
```

Comment:
This looks like an instance where Polymorphism and OOP would be better than the current approach because manualUpdate has rippling changes throughout the class and behaves differently for each case. 

In those situations, it's generally much easier to create 2 simple classes than one complex class. 

Something to consider refactoring for, after the demo.

Reply:
Great idea! And done. That was fun :)